test/Libraries/RevitIntegrationTests/SerializationTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.IO;
+using NUnit.Framework;
+using RevitTestServices;
+using RTF.Framework;
+using Revit.Elements;
+using System.Linq;
+using Autodesk.DesignScript.Geometry;
+using System;
+
+namespace RevitSystemTests
+{
+    [TestFixture]
+    class SerializationTests : RevitSystemTestBase
+    {
+        private string JsonFolderName = "DynamoRevitJsons";
+
+        [Test, TestModel(@".\empty.rfa")]
+        public void SaveXMLTestFilesToJson()
+        {
+            // Get all old XML DYNs
+            var di = new DirectoryInfo(workingDirectory);
+            var fis = new string[] { "*.dyn", "*.dyf" }
+            .SelectMany(i => di.GetFiles(i, SearchOption.AllDirectories))
+            .ToArray();
+            string[] XmlFilePaths = fis.Select(fi => fi.FullName).ToArray();
+
+            // Create new Folder under temp to store Json
+            var tempPath = Path.GetTempPath();
+            var jsonFolder = Path.Combine(tempPath, JsonFolderName);
+
+            if (!System.IO.Directory.Exists(jsonFolder))
+            {
+                System.IO.Directory.CreateDirectory(jsonFolder);
+            }
+
+            foreach (var filePath in XmlFilePaths)
+            {
+                // Open XMLs in new Dynamo Core which should take care of
+                // Migration but element binding might be lost
+                ViewModel.OpenCommand.Execute(filePath);
+                AssertNoDummyNodes();
+
+                // Get new file path under temp folder
+                var fileNameSameStructure = filePath.Split(new string[] { "\\test" }, StringSplitOptions.None).Last();
```

Comment:
this seems fine but I would use the system io path methods like combine if there is one available that suits your needs.

Reply:
Yeah, I did couple search and seems there is only way to get parent directory, `Path` class itself does not support split according to https://msdn.microsoft.com/en-us/library/system.io.path(v=vs.110).aspx. In this case, it can be nested pretty heavily so I chose to use `Split`